File: Python/BEGINNER/1182 Column in Array.py
# SUBPROGRAM
def makeMatrix():
    myMatrix = []
    for col in range(12):
        row = []
        for lin in range(12):
            row.append(float(input()))
        myMatrix.append(row)
    return myMatrix


def sumColumn(columnOp, matrix):
    columnSum = 0
    for lin in range(len(matrix)):
        columnSum += matrix[lin][columnOp]
    return columnSum


def averageColumn(columnOp, matrix):
    columnSum = sumColumn(columnOp, matrix)
    columnAverage = columnSum / len(matrix)
    return columnAverage


# MAIN
operationColumn = int(input())
operation = input()
thisMatrix = makeMatrix()
if operation == 'S':
    sumResult = sumColumn(operationColumn, thisMatrix)
    print('%.1f' % sumResult)
else:
    averageResult = averageColumn(operationColumn, thisMatrix)
    print('%.1f' % averageResult)

File: C/BEGINNER/1182 Column in Array.c
#include <stdio.h>
int main()
{
    int operationColumn, i, j;
    char operation[2];
    double matrix[12][12], sum;
    sum = 0;
    scanf("%d %s", &operationColumn, operation);
    for (i = 0; i < 12; i++)
        for (j = 0; j < 12; j++)
            scanf("%lf", &matrix[i][j]);
    for (i = 0; i < 12; i++)
        for (j = 0; j < 12; j++)
            if (j == operationColumn)
                sum += matrix[i][j];
    if (operation[0] == 'S')
        printf("%.1lf\n", sum);
    else
        printf("%.1lf\n", (sum / 12.0));
    return 0;
}